lib/salus/scanners/report_go_dep.rb

```diff
@@ -7,38 +7,106 @@
 module Salus::Scanners
   class ReportGoDep < Base
     def run
-      unless should_run?
+      if @repository.go_sum_present?
+        record_dep_from_go_sum
+      elsif @repository.go_mod_present?
+        record_dep_from_go_mod
+      elsif @repository.dep_lock_present?
+        record_dep_from_go_lock_package
+      else
         raise(
           InvalidScannerInvocationError,
-          'Cannot report on Go dependencies without a Gopkg.lock file.'
+          'Cannot report on Go dependencies without a Gopkg.lock, go.mod, or go.sum file'
         )
       end
+    end
+
+    def record_dep_from_go_sum
+      go_sum_path = "#{@repository.path_to_repo}/go.sum"
+      dep_list = []
+
+      File.open(go_sum_path).each("=\n") do |line|
+        dep_list.append({ "fullDependency" => line, "name" => get_name(line),
+         "version" => get_version(line) })
+      end
+
+      dep_list.each do |dependency|
+        record_dep_package(
+          name: dependency['name'],
+          reference: "N/A for go.mod/go.sum dependencies",
+          version_tag: dependency['version'],
+          dependency_file: "go.sum",
+          type: "go_sum"
+        )
+      end
+    end
+
+    def record_dep_from_go_mod
+      go_mod_path = "#{@repository.path_to_repo}/go.mod"
+      dep_list = []
+      parse_line = false
+
+      File.open(go_mod_path).each("\n") do |line|
```

Comment:
This is likely an edge case but for some go.mod files with only one package such as https://sourcegraph.cbhq.net/github.cbhq.net/c3/go-shamir/-/blob/go.mod it doesn't follow the `require (`. We may want to be careful there.

Reply:
File.foreach to avoid a slurp here too.  It's a minor nit as the go.sum likely won't grow unbounded but a little cleaner.

Reply:
Changed 👍